clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -448,7 +448,7 @@ private void maybeWaitForProducerId() {
                 transactionManager.transitionToFatalError(e);
                 break;
             } catch (IOException e) {
-                log.debug("Broker {} disconnected while awaiting InitProducerId response", e);
+                log.debug("Broker disconnected while awaiting InitProducerId response: {}", e);
```

Comment:
It's worth including the broker here as it was originally intended.

Reply:
I looked at the code in Sender.
Can I obtain broker thru transactionManager ?

I am new to this code.

Thanks

Reply:
`Node node = awaitLeastLoadedNodeReady(requestTimeoutMs);` should give you the `node`.